ADR/00xx-component-references.md

```diff
@@ -0,0 +1,147 @@
+  * In this case, the app images must be rebuilt for the intermediate image update to really be evaluated in a functional way.
+  * This is a many-to-one dependency. Many component images depend on one intermediate base image.
+* A user team has an **OLM operator**. When they propose an update to one of their operands with new code, they want to see if that’s going to break their operator. RHTAP should posit what a rebuild of the operator bundle will look like and if it passes its tests and report that feedback back to the original pull request that updated the code in one of the operand images ([RHTAP-992](https://issues.redhat.com/browse/RHTAP-992)). A real example of this is in [gatekeeper](https://github.com/gatekeeper/gatekeeper-operator) where the operator repo contains both the [controller code](https://github.com/gatekeeper/gatekeeper-operator/blob/main/controllers/gatekeeper_controller.go) and the [bundle metadata](https://github.com/gatekeeper/gatekeeper-operator/blob/main/config/manifests/bases/gatekeeper-operator.clusterserviceversion.yaml), which need to be built into separate images([1](https://github.com/gatekeeper/gatekeeper-operator/blob/main/Dockerfile) and [2](https://github.com/gatekeeper/gatekeeper-operator/blob/main/bundle.Dockerfile)), separate Components in RHTAP.
+
+  * In this case, the bundle image must be rebuilt for the operand image update to be really be evaluated in a functional way.
+  * This is a one-to-many dependency. One operator bundle depends on many operand images.
+* A user team just has **two components that depend on each other in a functional way**. When they propose an update to one component, they want to submit a corresponding change to the second component, and have those tested together before merging both.
+  * In this case, on any given day, component B’s PRs may or may not have dependencies on PRs from component A. Not all changes are linked.
+  * This is a many-to-many dependency. The user may have lots of components that depend on lots of other components.
+
+In theory **any combination of the cases above** could be present in an app. They could have one intermediate base image, many operand images that depend on that, and one or more operator bundles which depend on those.
+
+Today, users work around how complicated it is to manage digests themselves by instead using *floating tags*, which have the benefit of being easy to use - no need to update - but have the problem of being unclear. It's not exactly clear what you're building against if you refer to it by tag. Potentially insecure. We want to make it easy for users to do better.
+
+## Decision
+
+### API
+
+* Introduce a new CustomResource called `ComponentReference` that lets one Component declare that another Component **embeds a reference** to it.
```

Comment:
How will users be able to provide the data to AppStudio to fill in the required metadata for a ComponentReference? This won't happen with a PR itself -- that is a separate use case. Will this happen within the Tekton specification? Within the AppStudio UI? Directly with CRs?

Reply:
* :x: Will this happen within the Tekton specification?
* :heavy_check_mark: Within the AppStudio UI?
* :heavy_check_mark: Directly with CRs?